aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py

```diff
@@ -67,8 +72,11 @@ async def setup_es_connection():
 
 async def consume_logs(nw, inferenced_logs_queue):
     async def subscribe_handler(msg):
-        data = msg.data.decode()
-        await inferenced_logs_queue.put(pd.read_json(data, dtype={"_id": object, "cluster_id": str, "ingest_at": str}))
+        data = msg.data
+        log_payload_list = payload_pb2.PayloadList()
+        logs_df = pd.DataFrame(json_format.MessageToDict(log_payload_list.FromString(data))["items"])
```

Comment:
Create dataframes directly from messages instead of converting to json first

Reply:
You should generate code using [python-betterproto](https://github.com/danielgtaylor/python-betterproto) instead of the default python generator. It will generate dataclasses that will be much easier to use.